Code for Breast Cancer data set for final project AAI 500 at USD
Authors:     
Angelo Walsh   



In [59]:
import pandas as pd


In [60]:
# import cancer data set
cancer_df = pd.read_csv('Breast_Cancer_METABRIC_Epic_Hospital.csv', sep=',')
cancer_df.head(1)

,Patient ID,Age at Diagnosis,Type of Breast Surgery,Cancer Type,Cancer Type Detailed,Cellularity,Chemotherapy,Pam50 + Claudin-low subtype,Cohort,ER status measured by IHC,...,Overall Survival Status,PR Status,Radio Therapy,Relapse Free Status (Months),Relapse Free Status,Sex,3-Gene classifier subtype,Tumor Size,Tumor Stage,Patient's Vital Status
0,MB-0000,75.65,Mastectomy,Breast Cancer,Breast Invasive Ductal Carcinoma,High,No,claudin-low,1.0,Positve,...,Living,Negative,Yes,138.65,Not Recurred,Female,ER+/HER2- High Prolif,22.0,2.0,Living


In [61]:
# Clean the data
cancer_df = cancer_df.drop(columns=['Patient ID'])  # drop Patient ID column
cancer_df.columns = (   # normalize column names. Set all column names to lowercase, replace spaces with underscores, and remove special characters
    cancer_df.columns
    .str.strip()
    .str.lower()
    .str.replace(r"\s+", "_", regex=True)
    .str.replace(r"[^a-zA-Z0-9_]", "", regex=True)
    .str.replace(r"_+", "_", regex=True)
    .str.strip("_")
)
cancer_df.head(1)

,age_at_diagnosis,type_of_breast_surgery,cancer_type,cancer_type_detailed,cellularity,chemotherapy,pam50_claudinlow_subtype,cohort,er_status_measured_by_ihc,er_status,...,overall_survival_status,pr_status,radio_therapy,relapse_free_status_months,relapse_free_status,sex,3gene_classifier_subtype,tumor_size,tumor_stage,patients_vital_status
0,75.65,Mastectomy,Breast Cancer,Breast Invasive Ductal Carcinoma,High,No,claudin-low,1.0,Positve,Positive,...,Living,Negative,Yes,138.65,Not Recurred,Female,ER+/HER2- High Prolif,22.0,2.0,Living


In [62]:
# Set age column to integer type
cancer_df['age_at_diagnosis'] = cancer_df['age_at_diagnosis'].astype(int)  # set age column to integer type
cancer_df.head(1)


,age_at_diagnosis,type_of_breast_surgery,cancer_type,cancer_type_detailed,cellularity,chemotherapy,pam50_claudinlow_subtype,cohort,er_status_measured_by_ihc,er_status,...,overall_survival_status,pr_status,radio_therapy,relapse_free_status_months,relapse_free_status,sex,3gene_classifier_subtype,tumor_size,tumor_stage,patients_vital_status
0,75,Mastectomy,Breast Cancer,Breast Invasive Ductal Carcinoma,High,No,claudin-low,1.0,Positve,Positive,...,Living,Negative,Yes,138.65,Not Recurred,Female,ER+/HER2- High Prolif,22.0,2.0,Living


In [63]:
# Check for missing data
print("Original rows:", len(cancer_df))
print("Rows after dropna:", len(cancer_df.dropna()))
print("Rows lost:", len(cancer_df) - len(cancer_df.dropna()))
rows_lost_percentage = (len(cancer_df) - len(cancer_df.dropna())) / len(cancer_df) * 100
print("Percentage of rows lost:", round(rows_lost_percentage, 4))

Original rows: 15054
Rows after dropna: 14886
Rows lost: 168
Percentage of rows lost: 1.116


There are 168 rows with missing data. This represents 168 / 15054 = 1.116% of the original data.

In [64]:
# Summarize missing data
missing_summary = pd.DataFrame({
    "missing_count": cancer_df.isna().sum(),
    "missing_percent": (cancer_df.isna().mean() * 100).round(2)
})

missing_summary[missing_summary["missing_count"] > 0]

,missing_count,missing_percent
type_of_breast_surgery,6,0.04
cellularity,30,0.20
pam50_claudinlow_subtype,6,0.04
cohort,6,0.04
tumor_other_histologic_subtype,6,0.04
primary_tumor_laterality,18,0.12
lymph_nodes_examined_positive,6,0.04
mutation_count,60,0.40
nottingham_prognostic_index,6,0.04
pr_status,6,0.04


Columns with missing data have very low percentages, less than 0.4%. Missing percent for mutation count is 0.4, the next is cellularity at 0.2, the rest are even lower. 

In [65]:
# Drop missing data
cancer_df_cleaned = cancer_df.dropna()
print("Rows after dropping missing data:", len(cancer_df_cleaned))

Rows after dropping missing data: 14886


In [66]:
# Export cleaned data set
cancer_df_cleaned.to_csv("cancer_data_cleaned.csv", index=False)
print("Cleaned data exported to cancer_data_cleaned.csv")
print("Number of rows in cleaned data:", len(cancer_df_cleaned))
print("Number of columns in cleaned data:", len(cancer_df_cleaned.columns))

Cleaned data exported to cancer_data_cleaned.csv
Number of rows in cleaned data: 14886
Number of columns in cleaned data: 33
